server/grpc_service.go

```diff
@@ -350,15 +371,12 @@ func watchTSDeadline(ctx context.Context, tsDeadlineCh <-chan deadline) {
 
 // Bootstrap implements gRPC PDServer.
 func (s *GrpcServer) Bootstrap(ctx context.Context, request *pdpb.BootstrapRequest) (*pdpb.BootstrapResponse, error) {
-	forwardedHost := getForwardedHost(ctx)
-	if !s.isLocalRequest(forwardedHost) {
-		client, err := s.getDelegateClient(ctx, forwardedHost)
-		if err != nil {
-			return nil, err
-		}
-		ctx = grpcutil.ResetForwardContext(ctx)
+	fn := func(ctx context.Context, client *grpc.ClientConn) (interface{}, error) {
 		return pdpb.NewPDClient(client).Bootstrap(ctx, request)
 	}
+	if rsp, err := s.unaryMiddleware(ctx, request.GetHeader(), fn); err == nil && rsp != nil {
```

Comment:
We should handle the cases when the error is not nil.

Reply:
yes , it will return err if err is not nil